ui/components/app/assets/nfts/nfts-items/nfts-items.tsx

```diff
@@ -0,0 +1,300 @@
+        Object.keys(nftsDropdownState?.[selectedAddress]?.[chainId]).length ===
+          0)
+    ) {
+      const initState: Record<string, unknown> = {};
+      collectionsKeys.forEach((key: string) => {
+        initState[key] = true;
+      });
+
+      const newNftDropdownState = {
+        ...nftsDropdownState,
+        [selectedAddress]: {
+          ...nftsDropdownState?.[selectedAddress],
+          [chainId]: initState,
+        },
+      };
+
+      dispatch(updateNftDropDownState(newNftDropdownState));
+    }
+  }, [
+    collectionsKeys,
+    previousCollectionKeys,
+    nftsDropdownState,
+    selectedAddress,
+    chainId,
+    dispatch,
+  ]);
+
+  const getAssetImageUrlAndUpdate = async (
+    image: string | undefined,
+    nft: NFT,
+  ) => {
+    const nftImage = await getAssetImageURL(image, ipfsGateway);
+    const updatedNFt = {
+      ...nft,
+      ipfsImageUpdated: nftImage,
+    };
+    return updatedNFt;
+  };
+
+  useEffect(() => {
+    const promisesArr: Promise<NFT>[] = [];
+    const modifyItems = async () => {
+      for (const key of collectionsKeys) {
+        const { nfts } = collections[key];
+        for (const singleNft of nfts) {
+          const { image, imageOriginal } = singleNft;
+
+          const isImageHosted =
+            image?.startsWith('https:') || image?.startsWith('http:');
+          if (!isImageHosted) {
+            promisesArr.push(
+              getAssetImageUrlAndUpdate(imageOriginal ?? image, singleNft),
+            );
+          }
+        }
+      }
+      const settled = await Promise.all(promisesArr);
+      setUpdatedNfts(settled);
+    };
+
+    modifyItems();
+  }, []);
+
+  const history = useHistory();
+
+  const onSendNft = async (nft: NFT) => {
+    trackEvent(
+      {
+        event: MetaMetricsEventName.sendAssetSelected,
+        category: MetaMetricsEventCategory.Send,
+        properties: {
+          is_destination_asset_picker_modal: false,
+          is_nft: true,
+        },
+        sensitiveProperties: {
+          ...sendAnalytics,
+          new_asset_symbol: nft.name,
+          new_asset_address: nft.address,
+        },
+      },
+      { excludeMetaMetricsId: false },
+    );
+    await dispatch(
+      updateSendAsset({
+        type: AssetType.NFT,
+        details: nft,
+        skipComputeEstimatedGasLimit: false,
+      }),
+    );
+    history.push(SEND_ROUTE);
+    onCloseModal && onCloseModal();
+  };
+
+  const renderCollection = ({
+    nfts,
+  }: Collection & { key: string; isPreviouslyOwnedCollection: boolean }) => {
+    if (!nfts.length) {
+      return null;
+    }
+    const getSource = (isImageHosted: boolean, nft: NFT) => {
+      if (isImageHosted) {
+        return nft.image || undefined;
+      }
+
+      const found = updatedNfts.find(
+        (elm) =>
+          elm.tokenId === nft.tokenId &&
+          isEqualCaseInsensitive(elm.address, nft.address),
+      );
+
+      return found?.ipfsImageUpdated || undefined;
+    };
```

Comment:
This should also be removed in the follow up PR for this ticket: https://consensyssoftware.atlassian.net/browse/MMASSETS-498

Reply:
In fact, this entire component can probably be removed.